### PR TITLE
chore: generate typescript types from Golang models

### PR DESCRIPTION
**Pull request by Emyrk**
Sync FE and BE types. Github action to ensure they stay in sync.

### CODE REVIEW

File: .github/workflows/gen.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Ensure all generated files are up-to-date
+name: GenerateFiles
```
Comment (Emyrk) [inline .github/workflows/gen.yaml:2]: I think this action works... I did not test it.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+.PHONY: gen
```
Comment (Emyrk) [inline Makefile:1]: Time to learn Make!

File: go.mod [hunk 3]
```diff
@@ -1,23 +1,31 @@
 module github.com/JollyGrin/postgres-attendance
 
-go 1.22
+go 1.23.0
 
-toolchain go1.22.9
+toolchain go1.23.4
```
Comment (Emyrk) [inline go.mod:5]: Is this chill? Can you update your Go?

File: scripts/check_unstaged.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
```
Comment (Emyrk) [inline scripts/check_unstaged.sh:1]: _borrowed_ from https://github.com/coder/coder/blob/main/scripts/check_unstaged.sh

**Review by JollyGrin — APPROVED** [review]

**Pull request merged**